File: 4-go/models/product.go
package models

import "gorm.io/gorm"

type Product struct {
	gorm.Model
	ID          int64   `gorm:"primaryKey" json:"id"`
	Name        string  `json:"name" gorm:"column:name"`
	Category    int     `json:"category" gorm:"column:category"`
	Price       float32 `json:"price" gorm:"column:price"`
	CreatedDate string  `json:"createdDate" gorm:"column:created_date"`
	Available   int     `json:"available" gorm:"column:available"`
}

func (Product) TableName() string {
	return "product"
}

File: 3-kotlin/ktor-pkuszn-bot/src/main/kotlin/com/example/Product.kt
package com.example

import kotlinx.serialization.Serializable

@Serializable
public data class Product(val id: Int, val name: String, val category: Int, val price: Float, val createdDate: String, val available: Boolean) {
    fun toDiscordMessage(): String {
        return "Product ID: $id\nName: $name\nCategory: $category\nPrice: $price\nCreated Date: $createdDate\nAvailable: $available"
    }
}
